### ISSUE TITLE
License

### ISSUE THREAD

**Issue by csswizardry** [inuitcss/inuitcss#3]
Many moons ago, I picked the Apache 2.0 license, because it felt nice and permissive whilst also keeping full credit for myself. Are people still happy with Apache 2.0, or would a move to the more ubiquitous MIT license make sense?

---

**MIT – good**
- Nice and permissive
- Copyright gets maintained – license information has to stay in the code
- Fairly ubiquitous – people are used to it
- Zero liability – I don’t get in trouble if my clearfixes take your app offline

**MIT – bad**
- Modifications – People don’t need to explain how, when, why, where they modified original code

---

**Apache 2.0 – good**
- All as above
- Modifications – people have to point out exactly where they made any changes to the original code

**Apache 2.0 – bad**
- Prevents trademarking – license explicitly stops me from trademarking any of the source code

---

Does anybody else have any thoughts?

(For transparency and openness, I’m leaning toward keeping with Apache 2.0.)

**Comment by kaifaust** [comment]
Hi Harry! Big fan here. Your CSS work has taught me a lot. I really appreciate your contribution to the CSS world :) Now, I have a real question regarding this topic.

I'm working on a fork of inuitcss that is aimed to better fits the needs of my consultancy. It's heavily modified and is intended to be our starter kit for new client projects. To what extent should I point out changes to remain compliant/keep you happy? I am, of course, including credits/links in files that include original inuitcss code.

**Comment by csshugs** [comment]
@kaifaust 

[Here](http://www.apache.org/licenses/LICENSE-2.0#redistribution) it says, you have to explicitly mark the code, that you modified:

> a. You must give any other recipients of the Work or Derivative Works a copy of this License; and
b. **You must cause any modified files to carry prominent notices stating that You changed the files**; and
c. You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works; and
d. [...]

**Comment by csshugs** [comment]
@kaifaust 
Is your question answered, or do you have any further concerns about this?

**Comment by kaifaust** [comment]
Question answered, thanks!

**Comment by dudezilla80** [comment]
Hi, and good job on the framework! 

I'm working on a commercial project, and would like to comply 100% with the Inuit license.

Is the compiled CSS required to bear comments related to the additions that I made to the SCSS files? In this case, wouldn't the Apache 2.0 license get in the way of having minimized, optimized CSS that has all comments stripped in the production version?

Thanks, and keep on keepin' on!

**Comment by herzinger** [comment]
@dudezilla80 I would tackle it by inserting a small comment banner up top of the file after minification, with credits and a link to a txt, md or html page with the more detailed information. I already do that for general credits, licence and etc.

**Comment by herzinger** [comment]
@dudezilla80 I would tackle it by inserting a small comment banner up top of the file after minification, with credits and a link to a txt, md or html page with the more detailed information. I already do that for general credits, licence and etc.

**Comment by dudezilla80** [comment]
@herzinger Even if I do it like this, I think the Apache 2 license is too vague for this use case, and it leaves room for potential lawsuits. I think it would be a hassle for me to convince my management to use Inuit because of this. Wouldn't it be better either to have a clear commercial license, or to go all the way open source with an MIT license?

**Comment by herzinger** [comment]
@dudezilla80 I don't see in anyway how it is vague, sorry. Also, when building a commercial project on top of open code, we don't really get to choose the licencing of that code. You need to work around whatever licence is in place. 

Doing it in the way I suggested you'll only be vague if you don't document your changes and the due credits correctly, and then that would be on you. It's not that hard, and it's certainly easier than starting from scratch. 

Bottom line, following the licence requirements is the least you can do, since you'll be using code written by others for free in a commercial thing.

Also, seeing how inuitcss is a modular framework, you don't need to document removal ou addition of whole partials, I believe, so it's quite easy to track your changes.

**Comment by dudezilla80** [comment]
@herzinger I understand your point. But it's still confusing for me how to comply with this (quoting from the top post of this thread, by @csswizardry ):

> Modifications – people have to **point out exactly** where they made any changes to the original code

Assuming I generate CSS from SASS, autoprefix it, minify it and strip comments, the solutions that I see are the following:

- either I prepend to the resulting CSS a huge text that explains where **exactly** I made changes to the original code; very impractical, because I'd have to log all my development process there;

- or I prepend a link to the CSS that will point to the document where all the aforementioned literature is written.

Both cases imply maintaining a document that logs the changes to the framework for the whole duration of the project. My opinion is that, if this is true and I understood correctly the terms of the license, this way of working is highly impractical. I'd rather pay upfront a certain sum for the commercial framework than having to maintain a document such as that.

I'm really curious if anybody spotted the Inuit framework in the wild, in a commercial project which implements the Apache 2 license in a way that is satisfactory to the authors of the framework.

**Comment by herzinger** [comment]
The second option is what I am suggesting, and it's hardly impractical. I think you are most likely misunderstanding exactly what kind of modifications you would need to log.

Let's say you added inuitcss via npm or bower. If you are able to update or delete and install it again without losing anything, you probably don't need to point anything out. Whatever you made, it was built on top of the framework, or involved omitting parts of it, and those are the intended ways of working with inuitcss, since it's a modular scalable "bare-bones" framework.

**Comment by herzinger** [comment]
The licence's modifications clause applies only to modifications of the framework itself, not to code you write yourself on top of it.

**Comment by dudezilla80** [comment]
OK, then. I'll try working around these issues as well as I can. But my suggestion still stands for the authors: it would be much clearer for the users to have an MIT license like the majority of the known CSS frameworks (Bootstrap and Foundation comes to mind). Or, of course, go fully commercial.

P.S.: I'd still like to see an example of a company that implemented Inuit and fully complied with the license, in order to have a clear, working sample.

**Comment by herzinger** [comment]
The project I'm working on is private, and incomplete yet. So I can't publish the code, but I'll show you some snippets so you can have an idea:

#### style.css
```css
/*!
 * lejour-home - Landing-page of the web version of Lejour.
 * @author Lejour <contato@lejour.com.br> (https://lejour.com.br/)
 * @contributors SEE IN CONTRIBUTORS.md
 * @version v1.1.4
 * @link https://lejour.com.br/
 * @license SEE LICENSE IN LICENSE.md
 * @uses inuitcss (SEE DETAILS IN INUITCSS.md)
 *//*!*
```

#### INUITCSS.md
```markdown
# inuitcss

github.com/inuitcss | inuitcss.com

This project uses the ui-toolkit Lejour-Bootstrap, which is built on top 
of [inuitcss](http://github.com/inuitcss/inuitcss/), by [Harry Roberts](http://csswizardry.com/) ([@csswizardry](http://github.com/csswizardry/)), 
and follows his ITCSS philosophy. It also conforms to the BEMIT variation 
of the BEM naming convention. 

* [inuitcss](https://github.com/inuitcss/inuitcss)
* [ITCSS](https://www.youtube.com/watch?v=1OKZOV-iLj4)
* [BEMIT](goo.gl/WCt3Fa)
* [namespacing](http://goo.gl/WCt3Fa)

It's a slightly modified version of the original
framework, so this document logs the modifications made to it's core, 
in compliance with the Apache 2.0 license.


## Modifications

1. The layout object and widths utility classes were modified to use Avalon-gs,
by [Henrique Erzinger](http://herzinger.com.br/) ([@herzinger](http://github.com/herzinger/));
2. The spacing utility partial was edited to use the alternate naming convention
(AKA "hipster naming"), following [this issue](https://github.com/inuitcss/inuitcss/issues/159);

All other changes were made by omitting parts of the framework or adding 
completely new, unrelated partials, as is it's intended use.
```

**Comment by herzinger** [comment]
Also, this documentation is all by me, so I would like for @csswizardry to confirm if this is the way to go, and I can make any modifications needed. I also documented all changes in it's respective partials, even if it's lost in the build process, so it's visible in the git repository (even if it's private).

**Comment by mmzoo** [comment]
> I think I’m gonna stick with Apache 2.0. It’s very permissive, but will also keep everything ... pointing back to inuitcss :)

I might be misunderstanding something, but if the main goal is "pointing back to inuitcss", then MIT is on a par with Apache, isn't it?

> people have to point out exactly where they made any changes

How is this a benefit rather than introducing uncertainty and a gray area as @dudezilla80 validly pointed out?

And the fact that everybody here seeks individual confirmation on those gray areas by @csswizardry points into the direction of MIT.

Either way, thank you for the time you invest into this - It is certainly appreciated!

**Comment by herzinger** [comment]
>And the fact that everybody here seeks individual confirmation on those gray areas by @csswizardry points into the direction of MIT.

@mmzoo no, it just means we want to hear from the original author, and he's not active here for a long while. To be honest, I see no gray areas and I already have demonstrated a lot.

**Comment by dudezilla80** [comment]
@herzinger There are certainly gray areas. As a matter of fact, the company I'm working for didn't allow me to use InuitCSS because it was too risky for them legally wise (in the event that the author tried to make a "profit" from those gray areas).

**Comment by jtfairbank** [comment]
**License Choice**

Shoutout for changing it to full on GLPv3.0+.  That should make it clear, right? ;)  But for real, [Apache 2.0 is compatible with MIT](https://en.wikipedia.org/wiki/License_compatibility#/media/File:Floss-license-slide-image.png) and many backend devs have been using projects written in each for corporate work just fine.

**Examples Are Swell**

Honestly, most of the Licensing questions seem to be due to a lack of a clear example.  I find that many similar grassroots projects run into the same issue- it's easy to pick them up but the learning curve jumps around the point that you need to solve complex real world problems (including licensing).

**How I Can Contribute**

I was looking for a new CSS framework to play with for a side-project and discovered ITCSS last week... mind blown.  I also stubled across the [Kube Framework](https://github.com/imperavi/kube) which looked like the best of the "lightweight but functional-out-of-the-box" class.  As part of the side project, I'm hoping to create a fork which implements Kube in the InuitCSS / ITCSS style:

 1. Convert it to this layout.
 2. Separate out their components into objects and classes.
 3. Remove their superflous global variables and mixins, and follow the InuitCSS conventions for where to place them.
 4. ...
 5. Apply the Kube / InuitCSS mashup to a new iteration of my resume / portfolio site (which I'll open source).

I'm happy to pay extra attention to marking my changes in both `.scss` and `.css` files, as well as other license requirements.  I think both the conversion and open source portfolio site would serve as good examples for newcomers who may not realize that InuitCSS isn't, say, Bootstrap compatible right out of the box.  It'd also help people get up and running right away by providing a set of components to use.

Note that it is fine to [combine open source projects using different licenses](https://www.gnu.org/licenses/license-compatibility.html) as long as they are compatible and you keep both licenses intact.  Kube uses the MIT license, which is also a lax-permissive oss license and compatible with Apache 2.0.

**Help Me Help You**

  * If you have time, I'd appreciate the occasional Q&A via a github issue on this project (is there a better place to ask?).
  * It'd be sweet to get a code review at the end!  This'd also make sure I'm not leading others astray.
  * It'd be nice to have a companion repo such as "inuitcss-contrib" with a readme file pointing to quality resources: tutorials and blog posts, stand-alone objects, reusable components, framework conversions, example sites, and real-world usage.  A subdirectory in this repo with a README may also work, similar to [MomentJS's plugin documentation](http://momentjs.com/docs/#/plugins/).  Or you could go full crazy and do something like [Underscore Contrib](https://github.com/documentcloud/underscore-contrib).

**Comment by herzinger** [comment]
@jtfairbank I was about to point you to [this slack channel](https://herzinger.typeform.com/to/h6e3Km) I'm managing, but as I was writing this the notification of you joining popped up, so welcome hahaha

The example I posted some answers ago comes from a project similar to yours I did for a private company, so I'd be glad to help with anything you need, and even to become an active contributor to the project, since I'm also interested in the directions you want to go on.

**Comment by csshugs** [comment]
I'm guessing that everything regarding any license questions is clarified, so I close this.
